### PR TITLE
Added Assets and created book item component

### PR DESCRIPTION
**Pull request by nathan-hyan**
## Summary

Added book item card

### Android

image

#### iPhone 11
image

#### iPhone SE (or similar)
image

## Trello Card

https://trello.com/c/UEMdmDm8

### CODE REVIEW

File: wbooks/src/app/screens/BookList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { StyleSheet, View } from 'react-native';
+
+import BookItem from './components/BookItem';
+
+const BookList = () => (
+  <View style={styles.container}>
+    <BookItem />
+    <BookItem />
+    <BookItem />
+    <BookItem />
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/index.tsx:12]: Pass data via props, use `mockBook` (for example: `mockBook[0]`, `mockBook[1]`, etc...)

File: wbooks/src/app/screens/BookList/index.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  </View>
+);
+
+const styles = StyleSheet.create({
+  container: {
+    // flex: 1,
+  }
+});
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/index.tsx:20]: Split styles, also you can use here the styles that you declare in the `app/index` View

**Review by sfernandez11 — CHANGES_REQUESTED** [review]
Nice, some comments. Also, 2 comments about the assets:
1. We don't add all the assets at the same time, we add them when we need them (delete all the assets that you don't use now)
2. We don't organize by folder inside `@assets` (don't use `@assets/General` => use `@assets` directly)

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+/* eslint-disable no-use-before-define */
+import { Image, StyleSheet, Text, View } from 'react-native';
+import book from '@assets/General/img_book1.png';
+import { COLORS } from '@constants/colors';
+
+const functionName = () => (
```
[the remark this answers is not in the document]
Comment (nathan-hyan) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:7]: Sorry! I have a snippet for creating rn-function components wich by default names them as functionName hahaha.

File: wbooks/src/app/screens/BookList/index.tsx (hunk 1 above)
Comment (nathan-hyan) [reply wbooks/src/app/screens/BookList/index.tsx:12]: Replaced with temporal .map

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    shadowColor: COLORS.black,
+    shadowOffset: {
+      width: 0,
+      height: 1
+    },
+    shadowOpacity: 0.18,
+    shadowRadius: 1.0,
+    elevation: 1,
+    flexDirection: 'row',
+    flexWrap: 'wrap',
```
[the remark this answers is not in the document]
Comment (nathan-hyan) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:29]: You mean the "flexWrap" property? cause if i took anything else, it breaks

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  title: {
+    fontSize: 17,
+    color: COLORS.gray
+  },
+  author: {
+    fontSize: 15
+  },
+  text: {
+    marginLeft: 19
+    // flex: 2
```
[the remark this answers is not in the document]
Comment (nathan-hyan) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:46]: In marvel there's actually 20px of separation between text and the book img
image

[the remark this answers is not in the document]
Comment (nathan-hyan) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:46]: image

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    width: 335,
+    height: 90,
+    backgroundColor: COLORS.white,
+    marginBottom: 10,
+    paddingHorizontal: 30,
+    paddingVertical: 15
+  },
+  title: {
+    fontSize: 17,
+    color: COLORS.gray
```
[the remark this answers is not in the document]
Comment (nathan-hyan) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:39]: fontWeight not specified / modified on Marvel
image

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    shadowOpacity: 0.18,
+    shadowRadius: 1.0,
+    elevation: 1,
+    flexDirection: 'row',
+    flexWrap: 'wrap',
+    width: 335,
+    height: 90,
+    backgroundColor: COLORS.white,
+    marginBottom: 10,
+    paddingHorizontal: 30,
```
[the remark this answers is not in the document]
Comment (nathan-hyan) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:34]: image

There are 30px from the border to the book cover in Marvel

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (sfernandez11) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:39]: Check in the right panel (info: color, font (not necessary), size, weight, etc.)
image

[the remark this answers is not in the document]
Comment (nathan-hyan) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:39]: Yes, the typeface is different. I didn't realized that hahaha

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (sfernandez11) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:46]: You're right! My bad, keep `marginLeft: 20` and `flex: 1`

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (sfernandez11) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:29]: This is because the missing `flex: 1` or `numberOfLines={2}` I think 🤔

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (sfernandez11) [reply wbooks/src/app/screens/BookList/components/BookItem/index.tsx:34]: You're right! Today I woke up a bit crazy hahaha

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Image, Text, View } from 'react-native';
+import noBook from '@assets/img_book_placeholder.png';
+
+import { STYLES } from './styles';
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/components/BookItem/index.tsx:5]: `import styles from './styles';`

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+type Props = {
+  item: {
+    title: string;
+    author: string;
+    imageUrl: string | null;
+  };
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/components/BookItem/index.tsx:12]: Extract `Book` interface => 
`interface Props { item: Book };`
```
interface Book {
    title: string;
    author: string;
    imageUrl: string | null;
}
```
You'll gonna need it

File: wbooks/src/app/screens/BookList/components/BookItem/index.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+function BookItem(props: Props) {
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/components/BookItem/index.tsx:15]: You can destructure props => `function BookItem({ item }: Props) {`
To avoid repeating : `props.prop...`

File: wbooks/src/app/screens/BookList/components/BookItem/styles.ts [hunk 11]
```diff
@@ -0,0 +1,40 @@
+import { COLORS } from '@constants/colors';
+import { StyleSheet } from 'react-native';
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/components/BookItem/styles.ts:2]: Sort imports:
* global imports
* aliases
* local imports

File: wbooks/src/app/screens/BookList/components/BookItem/styles.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+export const STYLES = StyleSheet.create({
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/components/BookItem/styles.ts:4]: `export default StyleSheet.create({`

File: wbooks/src/app/screens/BookList/index.tsx [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import { BOOKS_MOCK } from '@constants/mockBooks';
+import React from 'react';
+import { View } from 'react-native';
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/index.tsx:3]: Sort imports

File: wbooks/src/app/styles.ts [hunk 14]
```diff
@@ -0,0 +1,11 @@
+import { COLORS } from '@constants/colors';
+import { StyleSheet } from 'react-native';
```
Comment (sfernandez11) [inline wbooks/src/app/styles.ts:2]: Sort

File: wbooks/src/app/styles.ts [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+
+export const STYLES = StyleSheet.create({
```
Comment (sfernandez11) [inline wbooks/src/app/styles.ts:4]: `export default StyleSheet.create({`

File: wbooks/src/app/screens/BookList/components/BookItem/styles.ts [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+    shadowOffset: {
+      width: 0,
+      height: 1
+    },
+    shadowOpacity: 0.18,
+    shadowRadius: 1.0,
+    elevation: 1,
+    flexDirection: 'row',
+    width: 335,
+    height: 90,
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/components/BookItem/styles.ts:17]: You forgot this comment

File: wbooks/src/app/screens/BookList/components/BookItem/styles.ts [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+    color: COLORS.gray,
+    fontSize: 15
+  },
+  text: {
+    marginLeft: 20,
+    flex: 1
+  },
+  image: {
+    paddingHorizontal: 20,
+    paddingVertical: 25
```
Comment (sfernandez11) [inline wbooks/src/app/screens/BookList/components/BookItem/styles.ts:38]: Why this change? For the assets it's correct to use `width` and `height` but isn't for the components/screens styles

**Review by sfernandez11 — CHANGES_REQUESTED** [review]
Last comments

**Review by sfernandez11 — APPROVED** [review]
🥇
